docs/contributor-guide/adr/0010-prefer-simplicity-to-optimization.md

```diff
@@ -0,0 +1,24 @@
+# 10. Prefer simplicity to optimization
```

Comment:
The content LGTM. I think just rephrasing the title to be a little more concrete/actionable is all that's needed here.

```suggestion
# 10. Confirm performance issues before optimizing
```

You should probably also update the filename to match.